### ISSUE TITLE
Timeout on expression fuzzer

### ISSUE THREAD

**Issue by Yuhta** [facebookincubator/velox#3870]
### Description

https://app.circleci.com/pipelines/github/facebookincubator/velox/19403/workflows/c2b92eeb-57e3-4d38-9e34-08b1338e2e90/jobs/122380/steps

### Error Reproduction

Seed: 3065220249

The command on the original CircleCI job:

```
#!/bin/bash -eo pipefail
mkdir -p /tmp/fuzzer_repro/
chmod -R 777 /tmp/fuzzer_repro
_build/debug/velox/expression/tests/velox_expression_fuzzer_test \
    --seed 123456 \
    --enable_variadic_signatures \
    --velox_fuzzer_enable_complex_types \
    --lazy_vector_generation_ratio 0.2 \
    --velox_fuzzer_enable_column_reuse \
    --velox_fuzzer_enable_expression_reuse \
    --duration_sec 60 \
    --logtostderr=1 \
    --minloglevel=0 \
    --repro_persist_path=/tmp/fuzzer_repro \
&& echo -e "\n\nFuzzer run finished successfully."
```

### Relevant logs

```bash
I0130 10:38:18.167838 153315 ExpressionFuzzer.cpp:903] ==============================> Started iteration 4885 (seed: 3065220249)
I0130 10:38:18.170123 153315 ExpressionFuzzer.cpp:654] Couldn't find any function to return 'ARRAY<DOUBLE>'. Returning a constant instead.
I0130 10:38:18.171737 153315 ExpressionFuzzer.cpp:654] Couldn't find any function to return 'ARRAY<BOOLEAN>'. Returning a constant instead.
I0130 10:38:18.171857 153315 ExpressionFuzzer.cpp:654] Couldn't find any function to return 'ARRAY<BOOLEAN>'. Returning a constant instead.
I0130 10:38:18.172385 153315 ExpressionFuzzer.cpp:654] Couldn't find any function to return 'MAP<BOOLEAN,DOUBLE>'. Returning a constant instead.

Too long with no output (exceeded 5m0s): context deadline exceeded
```

**Comment by pedroerp** [comment]
Last time I caught a similar issue it was due to fuzzer triggering an infinite loop in a string manipulation function. If there's a single thread burning 100% of a cpu, we can profile it a get to the actual buggy loop.

**Comment by bikramSingh91** [comment]
Looked into it this morning, it turns out the issue is in vector fuzzer, I tried to profile the call to generate complex constants as inputs and found that it sometimes takes upto 2-3 seconds for generating even 1 level of complex types (arrays and maps), more when generating nested complex types.

**Comment by pedroerp** [comment]
Same, it looks like the issue is that we multiply the inner vector size when generating complex vectors in the Fuzzer, so depending on how nested they are, the inner vector generated ends up being quite large. This specific case hangs because it's generating 10M UTF-8 strings. 

I'll add some way to limit how large these inner vectors can get when generating nested complex types.

**Comment by pedroerp** [comment]
This has been fixed in #3887
